Automattic/jetpack
# Implement disconnect_site function of Connection Manager and migrate IDC disconnect

Identity Crisis relies on implementing plugins to implement disconnect flows. This update implements the disconnect_site function stub in the Connection Manager based on Jetpack's implementation including stats/tracking. It then updates the jetpack_idc_disconnect callbacks to applicable modules so that if the IDC package is included/configured it doesn't need further implementation unless custom functionality is needed.

#### Changes proposed in this Pull Request:

* Implement disconnect_site and migrate jetpack_idc_disconnect logic into packages.

#### Does this pull request change what data or activity we track or use?


NO

#### Testing instructions:






Regression Testing

Verify that Identity Crisis flows are fully functional
-- Duplicate your JP connected site to a new site (update options to new url except leave jetpack_private_options untouched)
-- Perform an edit to a post (triggers sync)
-- Visit the Jetpack Dashboard
-- Safe Mode will be enabled and IDC options presented.
-- Verify that choosing this is a new site, doesn't modify the home_url or site_url of the existing cache site and instead the new site has its own cache site created and jetpack_private_options updated.
-- Verify that choosing this is the site has stats migrated and home_url and site_url of the existing cache site is updated.
Modify a site's site_url and home_url. Ensure these are synced to the WP.com Cache (unit tests cover)
Note: Quickest way to trigger IDC flows is to copy jetpack_private_options from Site A to Site B then trigger a content change on Site B (edit post, etc)

Explicit Testing

call disconnect_site( true ) and disconnect_site( false) from shell and ensure users are removed and that based on the option the site connection is retained or removed.
